src/install_pypi.rs

```diff
@@ -492,7 +494,27 @@ async fn uninstall_outdated_site_packages(site_packages: &Path) -> miette::Resul
             };
 
             if let Some(installed_dist) = installed_dist {
-                installed.push(installed_dist);
+                // Get the installer
+                let installer = installed_dist.installer();
+
+                // If we can't get the installer, we can't be certain that we have installed it
+                let installer = match installer {
+                    Ok(installer) => installer,
```

Comment:
can we do it here?
```
installer.unwrap_or_default() == PIXI_UV_INSTALLER {
                     installed.push(installed_dist);
                 }
```

Reply:
Ah well, not sure If I'm missing something but its a `Result<Option<String>>` where if there is an error its something other than the file could not be found :)

Reply:
> Ah well, not sure If I'm missing something but its a `Result<Option<String>>` where if there is an error its something other than the file could not be found :)

You mean this will not work?

```rust
 Ok(installer) => {
installer.unwrap_or_default() == PIXI_UV_INSTALLER {
                     installed.push(installed_dist);
  },
```

Reply:
That could but I don't mind it being split up personally. :)

Reply:
I agree with both